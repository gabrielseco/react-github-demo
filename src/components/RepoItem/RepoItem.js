// @flow
import React from 'react';
import styles from './RepoItem.scss';
import { GithubRepo } from '../../types';

type Props = {
  repo: GithubRepo
};

const RepoItem = ({ repo }: Props) => (
  <li className={`${styles.repoItem} repo-item`}>
    <a href={repo.svn_url} target="_blank" className={styles.title}>
      <p>{repo.name}</p>
    </a>
    {repo.description ? (
      <p className={styles.description}>{repo.description}</p>
    ) : (
      <p className={styles.description}>No description provided</p>
    )}
    <div className={styles.row}>
      {repo.language ? (
        <p className={styles.language}>{repo.language}</p>
      ) : null}
      <p className={styles.stars}>
        <span className={styles.star}>
          <svg
            aria-label="stars"
            viewBox="0 0 14 16"
            version="1.1"
            width="14"
            height="16"
            role="img"
          >
            <path
              fillRule="evenodd"
              d="M14 6l-4.9-.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14 7 11.67 11.33 14l-.93-4.74z"
            />
          </svg>
        </span>
        {repo.stargazers_count}
      </p>
    </div>
  </li>
);

export default RepoItem;
