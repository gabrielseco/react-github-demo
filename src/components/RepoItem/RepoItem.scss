$border-repo-item-color: #cccccc;
$padding-repo-item: 1.25em;
$title-repo-item-color: #0921ea;
$margin-repo-item: 20px 0;

.repoItem {
  border: 1px solid $border-repo-item-color;
  border-radius: 2px;
  font-family: 'Open Sans', sans-serif;
  margin: $margin-repo-item;
  padding: $padding-repo-item;
}

.title {
  color: $title-repo-item-color;
  display: block;
  margin: 0.625em 0;

  &:visited {
    color: $title-repo-item-color;
  }
}

.description {
  margin: 20px 0;
}

.row {
  display: block;
}

.language {
  display: inline;
  font-weight: bold;
  margin-right: 20px; 
}

.stars {
  display: inline;
}

.star {
  margin-right: 4px;
  vertical-align: middle;
}
