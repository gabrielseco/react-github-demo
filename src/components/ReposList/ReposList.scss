.reposList {
  list-style-type: none;
}
