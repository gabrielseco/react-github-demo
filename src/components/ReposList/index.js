export { default as ReposList } from './ReposList';
