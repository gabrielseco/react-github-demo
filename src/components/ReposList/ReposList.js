// @flow
import React, { Component } from 'react';
import { RepoItem } from './../RepoItem';
import styles from './ReposList.scss';
import { repos as reposMock } from './repos';
import { GithubRepo } from './../../types';

type Props = {
  repos: GithubRepo[]
};

class ReposList extends Component<Props> {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { repos } = this.props;
    const reposCollection = repos.map(repo => {
      return <RepoItem repo={repo} key={repo.id} />;
    });
    return <ol className={styles.reposList}>{reposCollection}</ol>;
  }
}

ReposList.defaultProps = {
  repos: reposMock
};

export default ReposList;
