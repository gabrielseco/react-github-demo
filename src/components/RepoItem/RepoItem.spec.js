import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import RepoItem from './RepoItem';
import { repos } from './../ReposList/repos';

describe('RepoItem suite', () => {
  it('should render the RepoItem component', () => {
    const comp = <RepoItem repo={repos[0]} />;
    const component = shallow(comp);
    const tree = renderer.create(comp).toJSON();
    expect(component).toBeDefined();
    expect(tree).toMatchSnapshot();
  });
});
