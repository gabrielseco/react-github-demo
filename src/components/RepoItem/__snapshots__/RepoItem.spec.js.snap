// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoItem suite should render the RepoItem component 1`] = `
<li
  className="repoItem repo-item"
>
  <a
    className="title"
    href="https://github.com/SaraVieira/30-30"
    target="_blank"
  >
    <p>
      30-30
    </p>
  </a>
  <p
    className="description"
  >
    30 React Components in 30ish days
  </p>
  <div
    className="row"
  >
    <p
      className="language"
    >
      JavaScript
    </p>
    <p
      className="stars"
    >
      <span
        className="star"
      >
        <svg
          aria-label="stars"
          height="16"
          role="img"
          version="1.1"
          viewBox="0 0 14 16"
          width="14"
        >
          <path
            d="M14 6l-4.9-.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14 7 11.67 11.33 14l-.93-4.74z"
            fillRule="evenodd"
          />
        </svg>
      </span>
      11
    </p>
  </div>
</li>
`;
